src/R/Components/Impl/Repl/Commands/SourceRScriptCommand.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.Common.Core.Shell;
+using Microsoft.R.Components.ContentTypes;
+using Microsoft.R.Components.Controller;
+using Microsoft.R.Components.Extensions;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.R.Package.Repl.Commands {
+    public sealed class SourceRScriptCommand : IMenuCommand {
+        private readonly IRInteractiveWorkflow _interactiveWorkflow;
+        private readonly IActiveWpfTextViewTracker _activeTextViewTracker;
+        private readonly bool _echo;
+
+        public SourceRScriptCommand(
+            IRInteractiveWorkflow interactiveWorkflow,
+            IActiveWpfTextViewTracker activeTextViewTracker,
+            bool echo
+        ) {
+            _interactiveWorkflow = interactiveWorkflow;
+            _activeTextViewTracker = activeTextViewTracker;
+            _echo = echo;
+        }
+
+        public CommandStatus Status {
+            get {
+                var status = CommandStatus.Supported;
+                if (_interactiveWorkflow.ActiveWindow == null) {
+                    status |= CommandStatus.Invisible;
+                } else if (!string.IsNullOrEmpty(GetFilePath())) {
+                    status |= CommandStatus.Enabled;
+                }
+                return status;
+            }
+        }
+
+        public CommandResult Invoke(object inputArg, ref object outputArg) {
```

Comment:
I wouldn't spend too much time on renaming at the moment. There is bunch of thing to review in commands, controllers and associated OLE shims. Probably can be done as part of dependencies rework task

On May 17, 2016, at 12:51, Alexander Sher <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/R/Components/Impl/Repl/Commands/SourceRScriptCommand.cs<https://github.com/Microsoft/RTVS/pull/1785#discussion_r63591395>:

> +            _echo = echo;
> +        }
> +
> +        public CommandStatus Status {
> +            get {
> +                var status = CommandStatus.Supported;
> +                if (_interactiveWorkflow.ActiveWindow == null) {
> +                    status |= CommandStatus.Invisible;
> +                } else if (!string.IsNullOrEmpty(GetFilePath())) {
> +                    status |= CommandStatus.Enabled;
> +                }
> +                return status;
> +            }
> +        }
> +
> +        public CommandResult Invoke(object inputArg, ref object outputArg) {


Do we really gain anything from having a sync version if we have async?
Understanding that this command is sync. Nothing else.

________________________________

In reply to: 63590758<https://github.com/Microsoft/RTVS/pull/1785#discussion_r63590758> <ancestors%20=%2063590758,63589496,63588197>

-
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/RTVS/pull/1785/files/00a3e2b87ac1f536f0dcd2ba5f85d49d99b43da9#r63591395>